diff --git a/node_modules/discord.io/lib/index.js b/node_modules/discord.io/lib/index.js
index b5ccf12..8e41dc1 100644
--- a/node_modules/discord.io/lib/index.js
+++ b/node_modules/discord.io/lib/index.js
@@ -2614,15 +2614,15 @@ function Member(client, server, data) {
 	this.id = data.user.id;
 	this.joined_at = Date.parse(data.joined_at);
 	this.color = colorFromRole(server, this);
+	var tempThis = this;
 	['username', 'discriminator', 'bot', 'avatar', 'game'].forEach(function(k) {
 		if (k in Member.prototype) return;
-
 		Object.defineProperty(Member.prototype, k, {
-			get: function() { return client.users[this.id][k]; },
-			set: function(v) { client.users[this.id][k] = v; },
-			enumerable: true,
+				get: function() { return client.users[tempThis.id][k]; },
+				set: function(v) { client.users[tempThis.id][k] = v; },
+				enumerable: true,
 		});
-	});
+});
 }
 function Role(data) {
 	copyKeys(data, this, ['permissions']);
